src/crimson/os/seastore/object_data_handler.cc

```diff
@@ -563,7 +561,7 @@ ObjectDataHandler::truncate_ret ObjectDataHandler::truncate(
 	return prepare_data_reservation(
 	  ctx,
 	  object_data,
-	  offset);
+	  p2roundup(offset, ctx.tm.get_block_size()));
```

Comment:
The two checks above need to be done on this aligned offset.

Reply:
the offset (op.extent.offset) truncate get is not aligned.   and which two checks need to be done here?

Reply:
```
 ObjectDataHandler::truncate_ret ObjectDataHandler::truncate(
   context_t ctx,
   objaddr_t offset)
 {
+  offset = p2roundup(offset, ctx.tm.get_block_size());
   return with_object_data(
     ctx,
     [this, ctx, offset](auto &object_data) {
       LOG_PREFIX(ObjectDataHandler::truncate);
       DEBUGT("truncating {}~{} offset: {}",
             ctx.t,
             object_data.get_reserved_data_base(),
             object_data.get_reserved_data_len(),
             offset);
```